data/model/_basequery.py

```diff
@@ -75,6 +80,29 @@ def get_existing_repository(namespace_name, repository_name, for_update=False, k
     return query.get()
 
 
+def get_existing_namespace_size(namespace_name):
```

Comment:
Speaking of existing namespaces, we will need some sort of backfill, or check to make sure that existing repositories with content that doesn't yet have their row in the `RepositorySize` be populated.

Reply:
It could be something like, on pushes, if the corresponding row doesn't exist, compute the size first. This would allow the push to proceed with updating the running count. 

This would make it easier avoid any possible drifts in the repository size, as opposed if it was done in a separate worker process.

Reply:
I agree that on push we need to calculate first if a repository size does not exist then follow up with the running size. When it comes to generic reporting we will fall back to calculating from the storage table if we do not have a repo size record.